lightbulb/app.py

```diff
@@ -1262,13 +1262,15 @@ def flatten_command_option(
             await event.interaction.create_response([])
             return
 
-        def convert_response_value(val: t.Union[str, int, float, hikari.CommandChoice]) -> hikari.CommandChoice:
+        def convert_response_value(
+            val: t.Union[str, int, float, hikari.impl.AutocompleteChoiceBuilder]
+        ) -> hikari.impl.AutocompleteChoiceBuilder:
             if isinstance(val, (str, int, float)):
-                return hikari.CommandChoice(name=str(val), value=val)
+                return hikari.impl.AutocompleteChoiceBuilder(name=str(val), value=val)
             return val
 
-        resp_to_send: t.List[hikari.CommandChoice] = []
-        if isinstance(response, (str, int, float, hikari.CommandChoice)):
+        resp_to_send: t.List[hikari.impl.AutocompleteChoiceBuilder] = []
```

Comment:
I would recommend changing all usage of `hikari.impl.AutocompleteChoiceBuilder` where it is being used as a type hint to `hikari.api.AutocompleteChoiceBuilder` in case anyone is using a custom `AutocompleteChoiceBuilder` that conforms to the api, but does not inherit from the impl.